### PR TITLE
Automated cherry pick of #110110: Fix issue that Audit Server could not correctly encode

### PR DESCRIPTION
**Pull request by sxllwx**
Cherry pick of #110110 on release-1.24.

#110110: Fix issue that Audit Server could not correctly encode

For: https://github.com/kubernetes/kubernetes/pull/110110#issuecomment-1314970859

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Comment by wojtek-t** [comment]
/assign 

Should this also be cherrypicked to earlier releases, given we're cherrypicking it?

/lgtm
/approve

@kubernetes/release-managers - for cherrypick approval

**Comment by xmudrii** [comment]
This generally looks good, but can you please confirm:

* Does this PR need to be cherry-picked to 1.23?
* Does this PR have a release note or is it supposed to be NONE (the parent PR has NONE)?

**Comment by sxllwx** [comment]
> This generally looks good, but can you please confirm:
> 
> * Does this PR need to be cherry-picked to 1.23?
> * Does this PR have a release note or is it supposed to be NONE (the parent PR has NONE)?

Thanks for the reminder, I've submitted another PR cherry-pick to release-1.23.

**Comment by xmudrii** [comment]
@sxllwx Thank you for the reminder!

> I've submitted another PR cherry-pick to release-1.23.
>
> https://github.com/kubernetes/kubernetes/pull/113918

Looks like you linked this PR instead of one for the release-1.23 branch. Is this the PR for the 1.23 branch: #113965? If that's the case, can we get it reviewed and approved?

> Of particular note are:
>
> Because PR https://github.com/kubernetes/kubernetes/pull/110110 brings an Issue https://github.com/kubernetes/kubernetes/issues/111985 fixed by PR https://github.com/kubernetes/kubernetes/pull/113133.
>
> For this reason, I cherry-pick the PR https://github.com/kubernetes/kubernetes/pull/113133 to release-1.23 as well as release-1.24 (release-note has been edited)

Is the order of merging PRs important in this case? That being said, should we first merge this PR or #113968?

**Comment by sxllwx** [comment]
> Looks like you linked this PR instead of one for the release-1.23 branch. Is this the PR for the 1.23 branch: https://github.com/kubernetes/kubernetes/pull/113965? If that's the case, can we get it reviewed and approved?

- emmmm....  cherry-pick #110110 for release-1.23 is https://github.com/kubernetes/kubernetes/pull/113965 (comment has been modified.)
- *can we get it reviewed and approved?* Yes.

> Is the order of merging PRs important in this case? That being said, should we first merge this PR or https://github.com/kubernetes/kubernetes/pull/113968?

We should merge in the following order:

- merge PR #113918 (this PR) first, then merge PR #113968
- merge PR #113965 first, then merge PR #113967

**Comment by xmudrii** [comment]
@sxllwx Let's get all PRs reviewed and approved, and then I can go ahead and cherry-pick-approve them. I assigned @wojtek-t on #113965.

**Comment by wojtek-t** [comment]
/lgtm
/approve
/retest

https://github.com/orgs/kubernetes/teams/release-managers - for cherrypick approval

**Comment by sxllwx** [comment]
@xmudrii PTAL thx.

**Comment by xmudrii** [comment]
I would like to see #113968 and #113967 LGTM-ed and approved before I approve this PR and #113965. I pinged some folks (@liggitt @sttts @deads2k) to take a look at #113968 and #113967. I'll approve all four PRs as soon as all PRs have LGTM + approved.

**Comment by liggitt** [comment]
> I pinged some folks (@liggitt @sttts @deads2k) to take a look at #113968 and #113967.

I reviewed/approved those.

**Comment by liggitt** [comment]
> > I pinged some folks (@liggitt @sttts @deads2k) to take a look at #113968 and #113967.
> 
> I reviewed/approved those, and those resolve an existing issue in 1.23/1.24, so I think those can merge ahead of this

(I also tested this combined with the other PR, and it worked properly, though the test changes in this PR this conflict with that PR and require a rebase after it merges)

**Comment by Verolop** [comment]
@sxllwx looks like this needs a rebase after https://github.com/kubernetes/kubernetes/pull/113968 was merged. Thanks!

**Comment by sxllwx** [comment]
> merged

Thank you for your guidance.

Already rebase, PTAL

**Review by xmudrii — APPROVED** [review]
Discussed on Slack: https://kubernetes.slack.com/archives/CJH2GBF7Y/p1673882647806869

/lgtm
/approve
/release-note-none
(from the parent PR)

**Pull request merged**
